services/wireguard/endpoint/kernelspace/client.go

```diff
@@ -65,18 +62,36 @@ func (c *client) ConfigureDevice(iface string, config wg.DeviceConfig, ipAddr ne
 	return c.wgClient.ConfigureDevice(iface, deviceConfig)
 }
 
-func (c *client) AddPeer(iface string, peer wg.AddPeerOptions, _ ...string) error {
+func (c *client) AddPeer(iface string, peer wg.Peer) error {
 	endpoint := peer.Endpoint
 	publicKey, err := stringToKey(peer.PublicKey)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "could not convert string key to wgtypes.Key")
+	}
+
+	// Apply keep alive interval
+	var keepAliveInterval *time.Duration
+	if peer.KeepAlivePeriodSeconds > 0 {
+		interval := time.Duration(peer.KeepAlivePeriodSeconds) * time.Second
```

Comment:
We do we storing `peer.KeepAlivePeriodSeconds` as int and converting to the `time.Duration` here? It can be `time.Duration` from the very beginning.